eviloauth/dispatcher.py

```diff
@@ -1,11 +1,80 @@
 import sys
 import logging
+from enum import Enum, auto
+from diskcache import Cache
 from eviloauth.idp import IDP
 from eviloauth.exceptions import EviloauthCommandException
+from eviloauth.access_token import AccessToken
+from eviloauth import app, cache, temp_dir, load_modules
+from datetime import datetime, timedelta
+from eviloauth.general_token import GeneralToken
+
+
+class CommandType(Enum):
+    EXIT = "exit"
+    MODULE = "module"
+    TOKENS = "tokens"
+    IDP = "idp"
+    TARGET = "target"
+    URL = "url"
+    HELP = "help"
+
+
+class TokenSubCommand(Enum):
+    LIST = "list"
+    DELETE = "delete"
+
+
+class IDPSubCommand(Enum):
+    LIST = "list"
+    CONFIGURE = "configure"
+
+
+class PlatformType(Enum):
+    AZURE = "azure"
+    AWS = "aws"
+    OKTA = "okta"
+
+
+class AzureAttack(Enum):
+    READ_MAIL = "read_mail"
+
+
+class TargetSubCommand(Enum):
+    LIST = "list"
+    SET = "set"
+
+
+class TokenManager:
+    def __init__(self, cache_dir, general_cache):
+
+        self.cache = Cache(cache_dir)
+        self.general_cache = general_cache
+
+    def list_tokens(self):
+        general_tokens = self.general_cache.get('tokens', {})
+        return list(general_tokens.keys())
+
+    def add(self, access_token):
+        if isinstance(access_token, AccessToken):
+            token_key = str(access_token)
+            self.cache.set(token_key, access_token)
+            return token_key
+
+    def delete(self, token_key):
+        general_tokens = self.general_cache.get('tokens', {})
+        if token_key in general_tokens:
+            del general_tokens[token_key]
+            self.general_cache.set('tokens', general_tokens)
+            return True
+        return False
+
+    def list(self):
```

Comment:
This is the same command but less complicates than list_tokens. What does list_tokens do differently?